test/integration/docker_test.go

```diff
@@ -158,3 +163,81 @@ func TestForceSystemdEnv(t *testing.T) {
+func TestDockerEnvContainerd(t *testing.T) {
+	if ContainerRuntime() != constants.Containerd {
+		t.Skip("skipping: TestDockerEnvContainerd can only be run with the containerd runtime")
+	}
+	profile := UniqueProfileName("dockerenv")
+	ctx, cancel := context.WithTimeout(context.Background(), Minutes(30))
+	defer CleanupWithLogs(t, profile, cancel)
+
+	// start the minikube with containerd runtime
+	args := append([]string{"start", "-p", profile}, StartArgs()...)
+	cmd := exec.CommandContext(ctx, Target(), args...)
+	startResult, err := Run(t, cmd)
+	if err != nil {
+		t.Errorf("failed to start minikube with args: %q : %v", startResult.Command(), err)
+	}
+	time.Sleep(time.Second * 10)
+
+	// if we are in a shell, we need to run 'ssh-agent bash' to enable the ssh-agent
+	// but if we are in an integration test, we don't have a bash, so we need to get the environment variables set by ssh-agent, and use them in the following tests
+	result, err := Run(t, exec.CommandContext(ctx, "ssh-agent"))
+	if err != nil {
+		t.Errorf("failed to execute ssh-agent bash, error: %v, output: %s", err, result.Output())
+	}
+	output := result.Output()
+	// get SSH_AUTH_SOCK
+	groups := regexp.MustCompile(`SSH_AUTH_SOCK=(\S*);`).FindStringSubmatch(output)
+	if len(groups) < 2 {
+		t.Errorf("failed to acquire SSH_AUTH_SOCK, output is %s", output)
+	}
+	sshAuthSock := groups[1]
+	// get SSH_AGENT_PID
+	groups = regexp.MustCompile(`SSH_AGENT_PID=(\S*);`).FindStringSubmatch(output)
+	if len(groups) < 2 {
+		t.Errorf("failed to acquire SSH_AUTH_SOCK, output is %s", output)
+	}
+	sshAgentPid := groups[1]
+
+	// execute 'minikube ssh-host --append-known'
+	result, err = Run(t, exec.CommandContext(ctx, Target(), "ssh-host", "--append-known", "-p", profile))
+	if err != nil {
+		t.Errorf("failed to execute 'minikube ssh-host --append-known', error: %v, output: %s", err, result.Output())
+	}
+
+	// execute 'minikube docker-env --ssh-host --ssh-add' and extract the 'DOCKER_HOST' environment value
+	cmd = exec.CommandContext(ctx, "/bin/bash", "-c", fmt.Sprintf("SSH_AUTH_SOCK=%s SSH_AGENT_PID=%s %s docker-env --ssh-host --ssh-add -p %s", sshAuthSock, sshAgentPid, Target(), profile))
+	result, err = Run(t, cmd)
+	if err != nil {
+		t.Errorf("failed to execute minikube docker-env --ssh-host --ssh-add, error: %v, output: %s", err, result.Output())
+	}
+
+	output = result.Output()
+	groups = regexp.MustCompile(`DOCKER_HOST="(\S*)"`).FindStringSubmatch(output)
+	if len(groups) < 2 {
+		t.Errorf("failed to acquire SSH_AUTH_SOCK, output is %s", output)
+	}
+	dockerHost := groups[1]
+
+	cmd = exec.CommandContext(ctx, "/bin/bash", "-c", fmt.Sprintf("SSH_AUTH_SOCK=%s SSH_AGENT_PID=%s DOCKER_HOST=%s docker version", sshAuthSock, sshAgentPid, dockerHost))
+
+	result, err = Run(t, cmd)
+	if err != nil {
+		t.Errorf("failed to execute 'docker version',error: %v, output:%s", err, result.Output())
```

Comment:
t.Fatalf since docker is not running no need to continue the test, also improve message to say, docker probably not runing needed for the test